src/v/redpanda/admin_server.cc

```diff
@@ -643,8 +668,8 @@ void admin_server::register_kafka_routes() {
 
           auto shard = _shard_table.local().shard_for(ntp);
           if (!shard) {
-              throw ss::httpd::not_found_exception(fmt::format(
-                "Topic partition {}:{} not found", topic, partition));
+              // This node is not a member of the raft group, redirect.
+              throw redirect_to_leader(*req, ntp);
```

Comment:
does `throw redirect_to_leader(*req, ntp);` compile? it looks like it is `throw ss::future<T>`. elsewhere, `throw co_await redirect_to_leader` made sense.